psygnal/_signal.py

```diff
@@ -913,3 +935,40 @@ def _is_subclass(left: AnyType, right: type) -> bool:
         if origin is Union:
             return any(issubclass(i, right) for i in get_args(left))
     return issubclass(left, right)
+
+
+def partial_weakref(partial_fun: PartialBoundMethod) -> Callable:
+    obj, name = _get_proper_name(partial_fun.func)
+    args_ = partial_fun.args
+    kwargs_ = partial_fun.keywords
+
+    def wrap(*args: Any, **kwargs: Any) -> Any:
+        ob = obj()
+        if ob is None:
+            return
+        getattr(ob, name)(*args_, *args, **kwargs_, **kwargs)
+
+    setattr(wrap, "__ref__", obj)
+
+    del partial_fun
+    return wrap
+
+
+def _get_proper_name(callback: MethodType) -> Tuple[weakref.ref, str]:
+    assert inspect.ismethod(callback)
+    obj = callback.__self__
+    if (
+        not hasattr(obj, callback.__name__)
+        or getattr(obj, callback.__name__) != callback
+    ):
+        # some decorators will alter method.__name__, so that obj.method
+        # will not be equal to getattr(obj, obj.method.__name__). We check
+        # for that case here and traverse to find the right method here.
+        for name in dir(obj):
+            meth = getattr(obj, name)
+            if inspect.ismethod(meth) and meth == callback:
+                return weakref.ref(obj), name
+        raise RuntimeError(
+            f"During bind method {callback} of object {obj} an error happen"
+        )
+    return weakref.ref(obj), callback.__name__
```

Comment:
```suggestion
def _get_proper_name(slot: MethodType) -> tuple[weakref.ref, str]:
    obj = slot.__self__
    # some decorators will alter method.__name__, so that obj.method
    # will not be equal to getattr(obj, obj.method.__name__). 
    # We check for that case here and find the proper name in the function's closures
    if getattr(obj, slot.__name__, None) != slot:
        for c in slot.__closure__ or ():
            cname = getattr(c.cell_contents, "__name__", None)
            if cname and getattr(obj, cname, None) == slot:
                return weakref.ref(obj), cname
        raise RuntimeError(
            f"Could not find method on {obj} corresponding to decorated function {slot}"
        )
    return weakref.ref(obj), slot.__name__
```

not critical here, since this is only during connection, but this is about 20x faster. (searching through the decorator's closures instead of `dir(class)`

Reply:
next nice thing which I do not know earlier.